keps/sig-auth/20190607-certificates-api.md

```diff
@@ -0,0 +1,285 @@
+owning-sig: sig-auth
+reviewers:
+  - TBD
+approvers:
+  - TBD
+editor: TBD
+creation-date: 2019-06-07
+last-updated: 2019-06-07
+status: implementable
+---
+
+# Certificates API
+
+## Table of Contents
+
+- [Certificates API](#certificates-api)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+  - [Design Details](#design-details)
+    - [Sequence of an Issuance](#sequence-of-an-issuance)
+    - [CertificateSigningRequest API Definition](#certificatesigningrequest-api-definition)
+    - [Automatic CSR Approval](#automatic-csr-approval)
+    - [Manual CSR Approval](#manual-csr-approval)
+    - [Automatic Signers](#automatic-signers)
+  - [Graduation Criteria](#graduation-criteria)
+    - [Beta -&gt; GA Graduation](#beta---ga-graduation)
+  - [Implementation History](#implementation-history)
+  - [Known Issues](#known-issues)
+
+## Summary
+
+The Certificates API enables automation of
+[x509](https://tools.ietf.org/html/rfc5280) credential provisioning by providing
+a programmatic interface for clients of the Kubernetes API to request and obtain
+x509 certificates from a Certificate Authority (CA).
+
+## Motivation
+
+The security of the Kubernetes platform is underpinned by a public key
+infrastructure (PKI). Each Kubernetes cluster has a root certificate authority.
+This CA is used to secure communication between cluster components. The
+Certificates API was originally merged to support [Kubelet TLS
+Bootstrap](https://github.com/kubernetes/community/blob/1fd524165bcf54d4bef99adb8332df72f4f88d5c/contributors/design-proposals/cluster-lifecycle/kubelet-tls-bootstrap.md) but users have also begun to use this API to 
+
+### Goals
+
+- Provide a generic API for PKI issuance to be consumed by both core Kubernetes
+  components as well as user workload running in the cluster.
+- Support extensions that allow for specialized certificate issuance.
+
+### Non-Goals
+
+- Build in support for specialized certificate issuance (e.g.
+  [LetsEncrypt](https://letsencrypt.org/)).
+
+## Proposal
+
+We intend to provision initial and renewed certificates in Kubernetes (often
+without operator intervention). The crux of this process is how to make (and
+automate) the decision of whether to approve or deny a particular certificate
+signing request.
+
+The role of a Registration Authority (referred to in this design as the
+approver) is to verify that a CSR satisfies two requirements:
+
+1. The subject of the CSR is the origin of the CSR. This addresses the threat of
+   a third party masquerading as an authorized subject.
+1. The subject of the CSR is authorized to act in the requested context. This
+   addresses the threat of an undesired subject joining the cluster.
+
+Iff these two requirements are met, the approver should approve the CSR and
+otherwise should deny the CSR. Once the CSR is approved a Certificate Authority
```

Comment:
Or ignore the CSR.
It should only be denied when approver is confident that the CSR falls under its jurisdiction and is invalid/malicious.

Also, maybe point out explicitly that there can be multiple approvers and signers, for different CAs and purposes.